### PR TITLE
[FEATURE] Infection Minigames with Non-Red and Blue Teams

### PR DESCRIPTION
**Pull request by Reed022**
## Purpose
Adjusts some legacy functionality of the Infection game mechanic to allow Infection minigames to use team colors other than red and blue. The Infection game mechanic settings have expanded to include 2 buttons for selecting the Infected and Survivor teams.

Previously, an Infection minigame needed a red and a blue team. With this change, an Infection minigame can have any color of teams. However, for it to function, there can only be two teams and they must match with the two teams specified in the game mechanic settings. 
Red and blue have remained the defaults for an Infection minigame, though. Old Infection games will not break.

### CODE REVIEW

**Review by FireInstall — APPROVED** [review]

**Comment by addstar** [comment]
@FireInstall is this fine to merge as it is right now? Will it cause any issues/conflicts with your dev work?

**Comment by FireInstall** [comment]
It's fine. We already had made a pull request in our own repo but decided to wait until this one can be merged in the official repo!
If there will be conficts I will merge them, it's gonna be fine 😄

**Comment by addstar** [comment]
So Reed... RGB colours next?? :joy:

**Review by addstar — APPROVED** [review]

**Pull request merged**
